docs/contrib/workflow.md

```diff
@@ -0,0 +1,232 @@
+(`git commit --amend`).
+
+If you will, you can have several commits in your branch but please use meaningful
+commit titles for them. First commit must describe a main idea of your pull request.
+Rest commit titles must describe changes which were added after the fist commit.
+
+Good commit titles in a branch:
+
+```
+Added HA flow create operation
+Added unit tests for HA flow create
+Fixed checkstyle issues
+Removed unused files
+```
+
+Bad commit titles in a branch:
+
+```
+Added HA flow create operation
+fix
+fix
+fix
+```
+
+```
+Added HA flow create operation
+Added HA flow create operation
+Added HA flow create operation
+```
+
+```
+Added HA flow create operation
+debug
+Fixed issues after code review
+```
+
+### Commit message
+
+Commit message consist of several parts:
+* Title - one sentence with describes the main idea of your changes
+* Description - several sentences which describes your changes and gives
+  some addition information
+* Tags - helps GitHub to link PRs and issues
+
+When you create a new pull request (PR) from your branch on GitHub it takes
+commit message of your first commit in a branch as PR description.
+GitHub can handle a set of tags from your commit message.
+
+Mostly we are using the following tags:
+
+1. `Closes #1234` - shows that your pull request closes GitHub issue #1234.
+   When your PR will be merged GitHub will close issue #1234 automatically
+2. `Related to #1234` - shows that your pull request related to GitHub issue
+   #1234 but do not close it. For example when you develop some big feature you can
+   have several PRs with different parts of your feature. Only last PR must have tag
+   `Closes`. Rest of your PRs must use tag `Related to` to do not close GitHub issue
+   when they will be merged.
+
+The previous paragraph says that you can have several commit in your PR. Only the first one
+must contain full commit message with title, description and tags. Rest commits
+in your branch can have only meaningful title. But remember that the best practice is to
+have only one commit per branch/PR to do not face with the problem of meaningless
+commit titles at all.
+
+## Review
+
+To show your changes to the other contributors you must
+[create](https://github.com/telstra/open-kilda/compare) a GinHub pull request.
+
+### GitHub pull request
+
+GitHub requires two branches for a pull request: base branch and your branch.
+Base branch for most of the PRs is `develop` branch.
+But if you want to split your changes on several PRs you can choose some other branch
+as a base.
+
+#### Examples 1:
+
+A developer has branch `feature/ha_flows` with a new feature and wants to create a PR.
+He/she creates new GitHub PR from branch `feature/ha_flows` with the base branch `develop`.
+A QA has branch `tests/ha_flows_func_tests` with tests for developer's feature. He/she
+creates new PR from branch `tests/ha_flows_func_tests` with the base branch
```

Comment:
I'd say "The developer" instead of "he/she"

The developer creates **a** new PR from **the** branch `tests/ha_flows_func_tests` with the base branch

Reply:
fixed